plugins/inventory/nb_inventory.py

```diff
@@ -943,6 +944,13 @@ def extract_cluster(self, host):
         except Exception:
             return
 
+    def extract_cluster_device(self, host):
+        try:
+            # cluster device does not have a slug
+            return host["device"]["name"]
+        except Exception:
```

Comment:
Is there a more specific exception we could use like KeyError?

Reply:
Could this also use a more common pattern of checking if the key exists before accessing, instead of a try/except block?

Reply:
I don't know if i have understand it right, but have developed a solution without try/catch and with returning a suitable value in every case.
New version see below